lib/galaxy/managers/landing.py

```diff
@@ -73,11 +74,22 @@ def create_workflow_landing_request(self, payload: CreateWorkflowLandingRequestP
             model.workflow_source = payload.workflow_id
         model.uuid = uuid4()
         model.client_secret = payload.client_secret
-        model.request_state = payload.request_state
+        model.request_state = self.validate_workflow_request_state(payload.request_state)
         model.public = payload.public
         self._save(model)
         return self._workflow_response(model)
 
+    def validate_workflow_request_state(self, request_state: dict) -> dict:
+        # This would ideally be run in the context of a workflow input definition
+        for key, value in request_state.items():
+            if isinstance(value, dict):
+                try:
+                    # persist values after model validators and aliases have been applied
+                    request_state[key] = DataOrCollectionRequestAdapter.validate_python(value)
```

Comment:
```suggestion
                    validated = DataOrCollectionRequestAdapter.validate_python(value)
                    request_state[key] = validated.model_dump()
```

seems to fix the error I've reported in the comment above ^ https://github.com/galaxyproject/galaxy/pull/21087#issuecomment-3412369482